Automattic/wp-calypso
# Welcome Tour: Add keyboard navigation via left/right arrow keys & Esc

#### Changes proposed in this Pull Request

Enables navigation in the Welcome Tour via left/right keys, and closing on Esc. Addresses #55634

Specs as in the issue:

- If the welcome tour is focused either by clicking on tabbing into it:
    - left/right arrow keys progress the slides
    - Esc minimizes the tour
- Certain elements in the editor may close when they lose focus e.g. block inserter dropdowns. This is noted in the issue and we may attempt to address as potential “core” issues instead.
- If the tour loses focus, then no navigation happens via keyboard events. The editor takes precedence and the tour remains open.

![Kapture 2021-09-09 at 15 16 17](https://user-images.githubusercontent.com/1705499/132684075-a2daf928-62d2-42db-b5a0-f6d2678aae23.gif)

#### Testing instructions



* Go to the editor, open the sidebar, and from there click on "Welcome Guide".
* Click or tab into the panel that opens to move the focus to it (anywhere should work).
* You can now navigate the slides using the left/right arrow keys and minimize using the Esc key.
* Expanding the tour again should enable navigation without additional clicking.

#### Known issues

* When initially opening the Welcome Guide via the sidebar OR by reloading the page with the tour open, the tour does not receive focus, hence it does not respond to keydown events. You will need to click or tab into anywhere in the tour to enable navigation via the keyboard. This is addressed in a separate issue: #55635
* The awkward navigation via tabbing + space/enter keys is noted/addressed in another issue: #56054



Fixes #55634

Props to @razvanpapadopol for the work/discussion/testing in the background!
